### PR TITLE
disabled the posibility to change the collection for workflow items to prevent errors

### LINKED ISSUE #2140: Workflow collection cannot be updated
**Describe the bug**
When the user edit a Workflow item and tries to update a collection an error appears.

**To Reproduce**
1. Login
2. Go to workflow items list section
3. Edit any workflow item
4. Try to change the collection.
5. You will recive an error 500 in the console:
image

These are the general header:
image

These are the parameters that was sent:
image

**Expected behavior**
User should be able to modify the collection for the workflow item

### PR DESCRIPTION
**Pull request by CrisGuzmanS**
Hello @tdonohue i have already disabled the posibility to change the collection for workflow items to prevent errors. 🥳 

## References
* Fixes #2140
* Fixes https://github.com/DSpace/DSpace/issues/7965

## Description
This is my approach to prevent that the user can change the collection for a workflowitem to prevent errors, because the modification for the workflowitem collection is not trivial.

## Instructions for Reviewers
1. Login as administrator
2. go to Workflowitems list
3. Edit any workflowitem
4. Try to change the collection (you should no be able to do it)
5. Create any item
6. Try to change the collection (you should be able to do it)

List of changes in this PR:
* I injected collectionModifiable in workflow-edit-page-routing.module.ts
* if collectionModifiable is not injected, that means that the collection will be modifiable.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by tdonohue** [comment]
@CrisGuzmanS : Thanks for the quick work here!  I just wanted to let you know, it appears these changes break some of our automated tests (`yarn test`).  I suspect the tests themselves need to be fixed / updated though... so, if you could look into fixing them in this PR that'd be the best scenario.

**Comment by CrisGuzmanS** [comment]
Thank you @tdonohue for your feedback, and sorry for those errors, i am struggling with my chrome binary from local, this is why i do not check the testing section in my pull request, so i am waiting for github to check if my test is passing. But i am going to work on this task, thank you Tim

**Comment by CrisGuzmanS** [comment]
Hello @tdonohue ✋ i already fixed the test issues 👍 .

File: src/app/submission/form/collection/submission-form-collection.component.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
             class="btn btn-outline-primary"
             (blur)="onClose()"
             (click)="onClose()"
-            [disabled]="(processingChange$ | async)"
+            [disabled]="(processingChange$ | async) || collectionModifiable"
```
Comment (tdonohue) [inline src/app/submission/form/collection/submission-form-collection.component.html:28]: "collectionModifiable=true" should mean that the collection *can* be modified (i.e. dropdown should be enabled).  But, here it's the opposite... "collectionModifiable=true" means that this dropdown is **disabled**.

I'd recommend we change this to be `|| !collectionModifiable` and then change `workflowitems-edit-page-routing.module.ts` to set it to `false`.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@CrisGuzmanS : This looks good overall and it works.  However, the variable "collectionModifiable" is not working as it is described.   I'd expect that setting "collectionModifiable=true" should mean that you can modify the collection (i.e. dropdown is enabled), while "collectionModifiable=false" should mean that you **cannot** modify the collection (i.e. dropdown is disabled).  Currently, you've implemented the opposite (see inline comment below), which I think looks confusing.

If you could simply switch the behavior of this setting, I think that'd make more sense.  Other than that, this code all works and looks good.

**Comment by CrisGuzmanS** [comment]
Hello @tdonohue i have already attended your feedback. Please, let me know if everything is ok

**Review by tdonohue — APPROVED** [review]
👍 Thanks @CrisGuzmanS !  This looks good now. Tested it and it works well too.

**Pull request merged**